db/changes.go

```diff
@@ -409,6 +419,10 @@ func (db *DatabaseCollectionWithUser) changesFeed(ctx context.Context, singleCha
 		//   2. A limit is specified on the incoming ChangesOptions, and that limit is reached
 		//   3. An error is returned when calling singleChannelCache.GetChanges
 		for {
+			if options.ChangesCtx.Err() != nil {
+				base.DebugfCtx(ctx, base.KeyChanges, "Terminating channel feed %s", base.UD(to))
+				return
```

Comment:
Are you intentionally not sending an error entry to the feed in this case, because the outer consumer of the feed is also terminating based ChangesCtx?

Reply:
That is correct, there are outside consumers that will exit the context has been canceled.

- https://github.com/couchbase/sync_gateway/blob/cb97fc90623afda7527556a80cebe251edc63f56/db/changes.go#L1510
- https://github.com/couchbase/sync_gateway/blob/cb97fc90623afda7527556a80cebe251edc63f56/rest/changes_api.go#L434 (this case weird because the context cancelation is not `ChangesOptions.ChangesCtx` directly, and a case for removal of the context as on `ChangesOptions`

In both cases `forceClose` is set if the client terminates the connection.